### PR TITLE
feat: add support for npm 7 workspaces

### PR DESCRIPTION
**Pull request by EricCrosson**
If there is no `lerna.json` manfiest, try reading a list of package
files or globs out of a `workspaces` key in the root `package.json`
manifest.

### CODE REVIEW

**Comment by EricCrosson** [comment]
cc @bitgopatmcl @ocornoc

File: src/lerna_manifest.rs [hunk 1]
```diff
@@ -17,21 +17,25 @@ struct LernaManifestFile {
     packages: Vec<String>,
```
Comment (bitgopatmcl) [inline src/lerna_manifest.rs:17]: (attaching comment to arbitrary line)

This file name is no longer accurate now that it supports npm.

File: src/lerna_manifest.rs [hunk 2]
```diff
@@ -65,21 +69,48 @@ where
...
+        let file = File::open(root.as_ref().join(Self::LERNA_MANIFEST_FILENAME))?;
         let reader = BufReader::new(file);
-        let manifest_contents: LernaManifestFile = serde_json::from_reader(reader)?;
-        Ok(LernaManifest {
-            internal_package_manifests: get_internal_package_manifests(&root, &manifest_contents)?,
+        let lerna_manifest_contents: LernaManifestFile = serde_json::from_reader(reader)?;
+        Ok(MonorepoManifest {
+            internal_package_manifests: get_internal_package_manifests(
+                &root,
+                lerna_manifest_contents.packages.iter(),
```
Comment (bitgopatmcl) [inline src/lerna_manifest.rs:86]: Will this work with:

```
"workspaces": [
  "packages/**"
],
```

File: src/lerna_manifest.rs [hunk 3]
```diff
@@ -65,21 +69,48 @@ where
...
+            )?,
+        })
+    }
+
+    pub fn from_directory<P>(root: P) -> Result<MonorepoManifest, Box<dyn Error>>
+    where
+        P: AsRef<Path>,
+    {
+        MonorepoManifest::from_lerna_manifest(&root)
+            .or_else(|_| MonorepoManifest::from_package_manifest(&root))
```
[the remark this answers is not in the document]
Comment (ericcrosson-bitgo) [reply src/lerna_manifest.rs:111]: Fair point. If we had both configured, the binary would use the `lerna.json`, which seems reasonable 

I can't think of a use-case of the top of my head where somebody would use both lerna and npm workspaces.  Maybe lerna and yarn workspaces? I'm not very familiar with yarn

File: src/lerna_manifest.rs (hunk 1 above)
Comment (ericcrosson-bitgo) [reply src/lerna_manifest.rs:17]: Fixed in 29a8999

File: src/lerna_manifest.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ericcrosson-bitgo) [reply src/lerna_manifest.rs:111]: Looks like there is a use-case for lerna and yarn workspaces, but the assumption is that both point to the same packages 
https://medium.com/@jsilvax/a-workflow-guide-for-lerna-with-yarn-workspaces-60f97481149d

**Review by ocornoc — APPROVED** [review]
This looks good: using either `package.json` or `lerna.json` to create the manifests.

File: src/lerna_manifest.rs (hunk 2 above)
Comment (bitgopatmcl) [reply src/lerna_manifest.rs:85]: It worked

**Pull request merged**